contracts/contracts/GrantRoundManager.sol

```diff
@@ -32,35 +24,70 @@ contract GrantRoundManager {
   /// @notice Address of the ERC20 token in which donations are made
   IERC20 public immutable donationToken;
 
+  /// @notice Used during donations to temporarily store swap output amounts
+  mapping(IERC20 => uint256) internal swapOutputs;
+
+  /// @notice Used during donations to temporarily store donation ratio sums to ensure totals are always 100%
+  mapping(IERC20 => uint256) internal donationRatios;
+
   /// @notice WETH address
-  address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+  IERC20 public constant WETH = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+
+  /// @notice Scale factor
+  uint256 internal constant WAD = 1e18;
```

Comment:
decimals here may lead to misleading amounts for tokens like USDC which only go to 1e6

Reply:
Hm, how so? Even if USDC amount where 1e6 =1 USDC, say you want 25% of 1 USDC, then you'd have `1e6 * 0.25e18 / 1e18 = 250000 = 0.25 USDC`

Reply:
right but what if someone tries to swap 0.2500009 USDC and submits that to the swap method on uni won't it revert?

Reply:
Do you have an example of how we might get such a value? When we divide by 1e18 at the end, we'll either be left 250,000 (0.25 USDC) or 250,001 (0.250001 USDC) because there's only 6 decimals of precision when do you `tokenDecimalsUsdc * xe18 / 1e18`